design/one-pager-default-resource-class.md

```diff
@@ -88,66 +118,308 @@ spec:
+its current state, which looks like this:
+
+```go
+// SetupWithManager adds a controller that reconciles MySQLInstance instance claims.
+func (c *MySQLInstanceClaimController) SetupWithManager(mgr ctrl.Manager) error {
+  r := resource.NewClaimReconciler(mgr,
+    resource.ClaimKind(databasev1alpha1.MySQLInstanceGroupVersionKind),
+    resource.ClassKind(v1alpha1.RDSInstanceClassGroupVersionKind),
+    resource.ManagedKind(v1alpha1.RDSInstanceGroupVersionKind),
+    resource.WithManagedConfigurators(
+      resource.ManagedConfiguratorFn(ConfigureMyRDSInstance),
+      resource.NewObjectMetaConfigurator(mgr.GetScheme()),
+    ))
+
+  name := strings.ToLower(fmt.Sprintf("%s.%s", databasev1alpha1.MySQLInstanceKind, controllerName))
+
+  return ctrl.NewControllerManagedBy(mgr).
+    Named(name).
+    Watches(&source.Kind{Type: &v1alpha1.RDSInstance{}}, &resource.EnqueueRequestForClaim{}).
+    For(&databasev1alpha1.MySQLInstance{}).
+    WithEventFilter(resource.NewPredicates(resource.HasClassReferenceKind(resource.ClassKind(v1alpha1.RDSInstanceClassGroupVersionKind)))).
+    Complete(r)
+}
+```
+To look as follows:
+
+```go
+// SetupWithManager adds a controller that reconciles MySQLInstance instance claims.
+func (c *MySQLInstanceClaimController) SetupWithManager(mgr ctrl.Manager) error {
+  r := resource.NewClaimReconciler(mgr,
+    resource.ClaimKind(databasev1alpha1.MySQLInstanceGroupVersionKind),
+    resource.ClassKinds{Portable: databasev1alpha1.MySQLInstanceGroupVersionKind, NonPortable: v1alpha1.RDSInstanceClassGroupVersionKind},
+    resource.ManagedKind(v1alpha1.RDSInstanceGroupVersionKind),
+    resource.WithManagedConfigurators(
+      resource.ManagedConfiguratorFn(ConfigureMyRDSInstance),
+      resource.NewObjectMetaConfigurator(mgr.GetScheme()),
+    ))
+
+  name := strings.ToLower(fmt.Sprintf("%s.%s", databasev1alpha1.MySQLInstanceKind, controllerName))
+
+  return ctrl.NewControllerManagedBy(mgr).
+    Named(name).
+    Watches(&source.Kind{Type: &v1alpha1.RDSInstance{}}, &resource.EnqueueRequestForClaim{}).
+    For(&databasev1alpha1.MySQLInstance{}).
+    WithEventFilter(resource.NewPredicates(resource.HasClassReferenceKind(mgr.GetClient(), resource.ClassKinds{Portable: databasev1alpha1.MySQLInstanceGroupVersionKind, NonPortable: v1alpha1.RDSInstanceClassGroupVersionKind}))).
+    Complete(r)
+}
 ```
 
-And a resource claim that defaulted to the resource class above could look like this in the general case:
+A claim referencing a portable class will now look as follows:
 
 ```yaml
-apiVersion: storage.crossplane.io/v1alpha1
-kind: NoSQLInstance
+apiVersion: database.crossplane.io/v1alpha1
+kind: MySQLInstance
 metadata:
-  name: dynamo-claim
-  namespace: demo
+  name: mysql-claim
 spec:
-  ...
+  classRef:
+    name: standard-mysql
+  writeConnectionSecretToRef:
+    name: rdsmysql
+  engineVersion: "5.6"
 ```
 
-Or in the more specific case:
+With an arbitrary number of potential portable `MySQLInstanceClass` objects that
```

Comment:
This line makes it seem like the claim above could map to an arbitrary number of potential portable class object, when it's classRef.name is standard-mysql so it will only map to a single portable class object in a given environment.

Suggest rewording to something like this: "A portable class can be referenced by name, and the portable class definition can be customized to use the cloud services a given application or team wants to use, e.g. RDS in one environment and CloudSQL in another."

Reply:
Updated in most recent commit. I didn't use the exact wording but I think I communicated the same sentiment.